OPAL/br/src/main/scala/org/opalj/br/fpcf/properties/alias/AliasEntity.scala

```diff
@@ -0,0 +1,117 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package br
+package fpcf
+package properties
+package alias
+
+import scala.runtime.ScalaRunTime
+
+import org.opalj.br.fpcf.properties.Context
+
+/**
+ * Represents a pair of [[AliasSourceElement]]s and a [[Context]] for each the elements to which an alias relationship can be assigned.
+ * It is used to query and store the associated alias property in the property store.
+ * The order of the elements is irrelevant, as the alias property is symmetric.
+ * To ensure this, the given elements might be swapped internally.
+ *
+ * @param e1 The first [[AliasSourceElement]] to which the alias relationship is assigned.
+ * @param e2 The second [[AliasSourceElement]] to which the alias relationship is assigned.
+ * @param c1 The [[Context]] for the first element in which the alias relationship is valid.
+ * @param c2 The [[Context]] for the second element in which the alias relationship is valid.
+ */
+class AliasEntity(
+    private val c1: Context,
+    private val c2: Context,
+    private val e1: AliasSourceElement,
+    private val e2: AliasSourceElement
+) {
+
+    /**
+     * A copy of two elements of this [[AliasEntity]].
+     * It is used to ensure that the order of the elements is irrelevant.
+     */
+    private[this] val (_element1, _element2) = (e1, e2) match {
+        case (e1, e2) if e1.hashCode() < e2.hashCode() => (e1, e2)
+        case (e1, e2)                                  => (e2, e1)
+    }
+
+    /**
+     * A copy of two contexts of this [[AliasEntity]]. It uses the same order as the elements.
+     */
+    private[this] val (_context1, _context2) = (c1, c2) match {
+        case (c1, c2) if e1 == _element1 => (c1, c2)
```

Comment:
Use `eq`, not `==`. `==` invokes the `equals` method in Scala instead of performing a reference equality check